### PR TITLE
Password length validation for sign up and pwd change

### PR DESCRIPTION
**Pull request by thi4go**
Passed password change validator module to validators folder

Closes https://github.com/decred/politeiagui/issues/115

### CODE REVIEW

File: src/validators/util.js [hunk 1]
```diff
@@ -17,11 +17,14 @@ export function proposalNameValidator(name, supportedCharacters) {
   return matches.length === name.length;
 }
 
+export function passwordLengthValidator(password, minLength) {
+  return password.length > minLength
```
Comment (sndurkin) [inline src/validators/util.js]: This should be `>=`.

File: src/validators/password-change.js [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
   if (!isRequiredValidator(values.existingPassword) || !isRequiredValidator(values.password) || !isRequiredValidator(values.password_verify)) {
     throw new SubmissionError({ _error: "All fields are required" });
   }
 
   if (!passwordVerifyValidator(values.password, values.password_verify)) {
     throw new SubmissionError({ _error: "Passwords do not match" });
   }
+
+  if (!passwordLengthValidator(values.password, policy.passwordminchars)){
+    throw new SubmissionError({ _error: "Password must be at least "+policy.passwordminchars+" digits." });
```
Comment (sndurkin) [inline src/validators/password-change.js]: Could you reword this to "Your password must be at least <policy.passwordminchars> characters."?

File: src/validators/signup.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+import { emailValidator, isRequiredValidator, passwordVerifyValidator, passwordLengthValidator } from "./util";
 
-const validate = values => {
+const validate = (policy, values) => {
   if (!isRequiredValidator(values.email) || !isRequiredValidator(values.password) || !isRequiredValidator(values.password_verify)) {
     throw new SubmissionError({ _error: "All fields are required" });
   }
 
+  if (!passwordLengthValidator(values.password, policy.passwordminchars)){
+    throw new SubmissionError({ _error: "Password must be at least "+policy.passwordminchars+" digits." });
```
Comment (sndurkin) [inline src/validators/signup.js]: Same as above.

**Comment by thi4go** [comment]
@sndurkin I implemented the changes, however I don't know why TravisCI is failing at that test... any ideas?

File: src/validators/util.js [hunk 4]
```diff
@@ -17,11 +17,14 @@ export function proposalNameValidator(name, supportedCharacters) {
   return matches.length === name.length;
 }
 
+export function passwordLengthValidator(password, minLength) {
+  return password.length >= minLength
```
Comment (oktapodia) [inline src/validators/util.js]: Can you add the semicolon, please?

File: src/components/PasswordChange/index.js [hunk 5]
```diff
@@ -4,11 +4,18 @@ import { SubmissionError } from "redux-form";
...
+
+  componentDidMount() {
+    this.props.policy || this.props.onFetchData();
+  }
+
   onChangePassword(props) {
-    validate(props);
+    const policy = this.props.policy;
+    
+    validate(props, policy);
```
Comment (oktapodia) [inline src/components/PasswordChange/index.js]: Because you already have the `policy` into the `props`, it should be great to keep the same function signature as before, please

File: src/validators/util.js (hunk 4 above)
Comment (thi4go) [reply src/validators/util.js]: Added ! sry about that

File: src/components/PasswordChange/index.js (hunk 5 above)
Comment (thi4go) [reply src/components/PasswordChange/index.js]: Actually, the props in this function is the argument of onChangePassword(props), which contains only the information of the Form. I could maybe add policy to props.policy, or something like that?

Comment (oktapodia) [reply src/components/PasswordChange/index.js]: oh! Sorry did not check that!

**Comment by thi4go** [comment]
Fixed Travis CI ! Thanks @oktapodia

**Comment by oktapodia** [comment]
You just forgot to refer the test [here](https://github.com/decred/politeiagui/blob/master/test/e2e/main/signup.js#L19) and we will be ready to go

**Comment by thi4go** [comment]
Done @oktapodia  !! Thanks for the review
